packages/search/addon/services/navi-search-provider.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * This service is used to discover all the available search providers.
+ */
+
+import Service from '@ember/service';
+import { getOwner } from '@ember/application';
+import config from 'ember-get-config';
+
+/* global requirejs */
+
+export default class NaviSearchProviderService extends Service {
+  /**
+   * @method getProvider
+   * @param name
+   * @returns {Object} search provider service object
+   */
+  getProvider(name) {
+    return getOwner(this).lookup(`service:navi-search/${name}`);
+  }
+
+  /**
+   * @method all
+   * @returns {Array} array of available search providers
+   */
+  all() {
+    const searchProvidersRegex = new RegExp(`^(?:${config.modulePrefix}/)?services/navi-search/([a-z-]*)$`),
+      searchProviderServices = Object.keys(requirejs.entries).filter(
+        requirejsFileName =>
+          searchProvidersRegex.test(requirejsFileName) && !requirejsFileName.includes('navi-base-search-provider')
```

Comment:
We could just exclude `navi-base-search-provider` in the regex

Reply:
or just move it up one level?

Reply:
moved it up one level

Reply:
Then we can remove the `!requirejsFileName.includes(...` check